web/src/flow/stages/captcha/CaptchaStage.ts

```diff
@@ -45,6 +45,16 @@ export class CaptchaStage extends BaseStage<CaptchaChallenge, CaptchaChallengeRe
     @state()
     scriptElement?: HTMLScriptElement;
 
+    @property({ type: Boolean })
+    embedded = false;
+
+    @property()
+    onTokenChange: (token: string) => void = (token) => {
```

Comment:
**This *is* a change request.**

I don't think the default value you set here does what you think it should do.  `onTokenChange` is a way for the client code to pass a mutator, a function that reaches _up_ into the client and sets a value there, as you're doing in the IdentificationStage.ts changes below.

You have two choices.  The first is 

```
onTokenChange: (token: string) => void = (_token: string) => undefined;
```

... which will swallow any errors until you realize that you forgot to supply `onTokenChange` in your client code.

The better choice is:

```
onTokenChange: (token: string) => void = (_token: string) => { 
     throw new Error("Client failed to supply a handling function to CaptchaStage"); 
};
```

... which will crash early and crash informatively if you do forget.  (Regarding my comment about arrow functions, a `throw` is always a statement, so it has to be wrapped in scope delimiters.)  

In either case, the function _declaration's_ argument signature must begin with an underscore (the `_token`), because it is a declaration and eslint will complain that all declared variables are not being used, but it is a convention that eslint should not make that complaint about variable names that begin with an underscore.  (The names in the _definition's_ signature are pure documentation, except when they're keys for a defined record object being passed in. TypeScript does have its little annoyances.)

Reply:
**This is not a change request. It is just a suggestion.**

Also?  That type-and-default pairing makes my head hurt.  Maybe you could, somewhere above the `class` declaration, put:

```
type TokenHandler = (token: string) => void;
```

... and then:

``` 
onTokenChange: TokenHandler = (_token: string) => { 
     throw new Error("Client failed to supply a handling function to CaptchaStage");
};
```